File: backend/src/main/java/com/observability/domain/entities/Alert.java
package com.observability.domain.entities;

import com.observability.domain.valueobjects.AlertSeverity;
import com.observability.domain.valueobjects.AlertStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.Map;

/**
 * Represents an alert in the system.
 * 
 * Alerts are triggered when certain conditions are met,
 * such as threshold violations or pattern detection.
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Alert {
    
    private String id;
    private String name;
    private String description;
    private AlertSeverity severity;
    private AlertStatus status;
    private String service;
    private Instant triggeredAt;
    private Instant resolvedAt;
    private Map<String, Object> metadata;
    private String ruleId;
    
    /**
     * Marks the alert as resolved.
     */
    public void resolve() {
        this.status = AlertStatus.RESOLVED;
        this.resolvedAt = Instant.now();
    }
    
    /**
     * Marks the alert as active.
     */
    public void activate() {
        this.status = AlertStatus.ACTIVE;
        this.triggeredAt = Instant.now();
    }
    
    /**
     * Validates if the alert has all required fields.
     * 
     * @return true if valid, false otherwise
     */
    public boolean isValid() {
        return name != null && !name.isBlank()
            && severity != null
            && status != null
            && triggeredAt != null;
    }
}

File: frontend/src/types/alert.ts
export enum AlertSeverity {
  LOW = 'LOW',
  MEDIUM = 'MEDIUM',
  HIGH = 'HIGH',
  CRITICAL = 'CRITICAL',
}

export enum AlertStatus {
  ACTIVE = 'ACTIVE',
  RESOLVED = 'RESOLVED',
  ACKNOWLEDGED = 'ACKNOWLEDGED',
  SUPPRESSED = 'SUPPRESSED',
}

export interface Alert {
  id: string
  name: string
  description?: string
  severity: AlertSeverity
  status: AlertStatus
  service?: string
  triggeredAt: string
  resolvedAt?: string
  metadata?: Record<string, unknown>
  ruleId?: string
}

export interface AlertRequest {
  name: string
  description?: string
  severity: AlertSeverity
  service?: string
  metadata?: Record<string, unknown>
  ruleId?: string
}

export interface AlertFilters {
  status?: AlertStatus
  severity?: AlertSeverity
  service?: string
  startTime?: string
  endTime?: string
  page?: number
  size?: number
}

